ml_genn/model.py

```diff
@@ -408,175 +375,319 @@ def traverse_tf_out_layers(out_layers):
 
-                        if isinstance(tf_in_layer, (
-                                tf.keras.layers.AveragePooling2D,
-                                tf.keras.layers.GlobalAveragePooling2D)):
-
-                            # traverse ignored layers to find more inputs
-                            pool_in_layers = traverse_tf_in_layers(tf_in_layers_all[tf_in_layer])
-
-                            # set pooling or global pooling
-                            if isinstance(tf_in_layer, tf.keras.layers.AveragePooling2D):
-                                pool_size = tf_in_layer.pool_size
-                                pool_strides = tf_in_layer.strides
-                                pool_padding = tf_in_layer.padding
-                            elif isinstance(tf_in_layer, tf.keras.layers.GlobalAveragePooling2D):
-                                pool_size = tf_in_layer.input_shape[1:3]
-                                pool_strides = None
-                                pool_padding = 'valid'
-
-                            # create connections for all pool layer inputs
-                            for pool_in_layer in pool_in_layers:
-                                sources.append(mlg_layer_lookup[pool_in_layer])
-                                synapses.append(AvePool2DDenseSynapses(
-                                    units=tf_layer.units,
-                                    pool_size=pool_size,
-                                    pool_strides=pool_strides,
-                                    pool_padding=pool_padding,
-                                    connectivity_type=connectivity_type))
-                                weights.append(tf_layer.get_weights()[0])
+                    assert(len(tf_in_layers) == 1)
+                    tf_in_layer = tf_in_layers[0]
+                    in_configs = configs_lookups[tf_in_layer]
+
+                    # configure layer
+                    config = LayerConfig()
+                    config.is_output = len(tf_out_layers) == 0
+                    config.name = tf_layer.name
+                    config.shape = tf_layer.output_shape[1:]
+                    config.neurons = converter.create_neurons(tf_layer, pre_compile_output)
+                    if not tf_layer.activation is tf.keras.activations.linear:
+                        config.has_activation = True
+
+                    converter.validate_tf_layer(tf_layer, config)
+
+                    # configure synapses
+                    for in_config in in_configs:
+
+                        if in_config.has_activation:
+                            # configure Dense synapses
+                            config.synapses_class.append(DenseSynapses)
+                            config.synapses_params.append({'units': tf_layer.units})
+                            config.synapses_source.append(in_config)
+                            config.synapses_weights.append(tf_layer.get_weights()[0])
```

Comment:
having to append stuff to four lists to add a synapse seems a bit ugly, maybe replace with a single list of ``collection.namedtuple`` e.g.
```python
InSynConfig = namedtuple('InSynConfig', ['class', 'params', 'source', 'weights'])
```
Then, this could be replaced by:
```python
config.synapses.append(InSynConfig(class=DenseSynapse, params={'units': tf_layer.units}, 
                                   source=in_config, weights=tf_layer.get_weights()[0]))
```